lib/protocol/protocol.js

```diff
@@ -352,7 +354,7 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
       if (isErrorType(err, errors.ProxyRequestError)) {
         protocolLogger.error(`Encountered internal error running command:  ${JSON.stringify(err)} ${err.stack}`);
         actualErr = err.getActualError();
-      } else if (!isW3C && (!(isErrorType(err, ProtocolError) || isErrorType(err, errors.BadParametersError)))) {
+      } else if (!isW3C && isUnknownError(err)) {
         protocolLogger.error(`Encountered internal error running command: ${err.stacktrace || err.stack}`);
         actualErr = new errors.UnknownError(err);
```

Comment:
can we now revert the change in `lib/jsonwp-proxy/proxy.js`? Perhaps the whole try/catch is not needed there now

Reply:
It can't be reverted, or _every_ proxied error will turn into an unknown error (see https://github.com/appium/appium-base-driver/blob/master/lib/protocol/protocol.js#L452-L458). Removing the whole catch in https://github.com/appium/appium-base-driver/blob/master/lib/jsonwp-proxy/proxy.js#L399-L405 might be possible.

Reply:
yes, lets remove the whole